test/unit/specs/core/edgeNetwork/injectProcessWarningsAndErrors.spec.js

```diff
@@ -13,63 +13,113 @@ governing permissions and limitations under the License.
-      },
-      getErrors() {
-        return [];
-      }
-    };
-    logger = jasmine.createSpyObj("logger", ["warn"]);
+    logger = jasmine.createSpyObj("logger", ["warn", "error"]);
     processWarningsAndErrors = injectProcessWarningsAndErrors({ logger });
   });
 
+  it("throws error if status code is below 2xx", () => {
+    expect(() => {
+      processWarningsAndErrors({
+        statusCode: 199
+      });
+    }).toThrowError(
+      "The server responded with a status code 199 and no response body."
+    );
+  });
+
+  it("throws error if status code is above 2xx", () => {
+    expect(() => {
+      processWarningsAndErrors({
+        statusCode: 300
+      });
+    }).toThrowError(
+      "The server responded with a status code 300 and no response body."
+    );
+  });
+
+  it("throws error if no parsed body and HTTP status code is not 204", () => {
+    expect(() => {
+      processWarningsAndErrors({
+        statusCode: 200
+      });
+    }).toThrowError(
+      "The server responded with a status code 200 and no response body."
+    );
+  });
+
+  it("throws an error if parsed body does not have handle array", () => {
+    expect(() => {
+      processWarningsAndErrors({
+        statusCode: 200,
+        body: '{"foo":"bar"}',
+        parsedBody: { foo: "bar" }
+      });
+    }).toThrowError(
+      'The server responded with a status code 200 and response body:\n{\n  "foo": "bar"\n}'
+    );
+  });
+
   it("logs warnings", () => {
-    response.getWarnings = () => [
+    const warnings = [
       {
-        code: "general:100",
-        message: "General warning."
+        title: "General warning",
+        detail: "General warning detail"
       },
       {
-        code: "personalization:204",
-        message: "Personalization warning."
+        title: "Personalization warning",
+        message: "Personalization warning detail"
       }
     ];
 
-    processWarningsAndErrors(response);
+    processWarningsAndErrors({
+      statusCode: 200,
+      parsedBody: {
+        handle: [],
+        warnings
+      }
+    });
 
     expect(logger.warn).toHaveBeenCalledWith(
-      "Warning received from server: [Code general:100] General warning."
+      "The server responded with a warning:",
+      warnings[0]
     );
     expect(logger.warn).toHaveBeenCalledWith(
-      "Warning received from server: [Code personalization:204] Personalization warning."
+      "The server responded with a warning:",
+      warnings[1]
     );
   });
 
-  it("throws errors", () => {
-    response.getErrors = () => [
+  it("logs non-fatal errors", () => {
+    const errors = [
       {
-        code: "general:100",
-        message: "General error occurred."
+        title: "General warning",
+        detail: "General warning detail"
       },
       {
-        code: "personalization:204",
-        message: "Personalization error occurred."
+        title: "Personalization warning",
+        message: "Personalization warning detail"
```

Comment:
We now use `title` and `detail`, instead of `message` -- is this a leftover?

Reply:
Technically, our source code no longer cares what the contents of the errors or warnings are. We just log the full object. In other words, it doesn't really matter what the warning and error objects look like in this spec. I'll update the tests though to either match the new warning and error object structure or perhaps deviate from it further to accentuate the point that the object structure doesn't matter.

Reply:
Oh, I see what you're pointing out now. I renamed one of the object's `title` property to `detail`, but not the `title` property on the other object. While the object structure doesn't actually matter, I do want these to be consistent and having them not consistent was an oversight. I'll update that. Thanks.